### ISSUE TITLE
LLVM: we need to update to use opaque pointers

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#900]
https://llvm.org/docs/OpaquePointers.html#frontends

> Frontends need to be adjusted to track pointee types independently of LLVM, insofar as they are necessary for lowering.

In practice this means all the `getElementType` need to go.

**Comment by czgdp1807** [comment]
Sure. We can do that. Depends on how urgently we need to do it and by when we need it. I already try to avoid `getElementType` in LLVM backend. We can remove the existing calls as well which will help in improving the ASR design.

**Comment by certik** [comment]
We have to do that, otherwise LFortran will not work with LLVM 15.

**Comment by czgdp1807** [comment]
Yes. When's the LLVM 15 releasing? Any due date so that I can prioritise things accordingly.

**Comment by certik** [comment]
It's already released and in Conda.

**Comment by czgdp1807** [comment]
Makes sense will do it tomorrow then.
